### DISCUSSION TITLE
App::update() in continuous mode

### DISCUSSION THREAD

**Discussion by aimer63** [emilk/egui#3562] [Q&A]
I'm writing a messaging app, the app manages a websocket connection ([ewebsock](https://github.com/rerun-io/ewebsock)) from the update() function. I need to run in continuous mode in order to handle messages even when the application is not active (no focus, no UI activities, window in background or minimized). I understood I can achieve this behavior calling ```egui::Context::request_repaint()``` from ```App::update()``` function. 
The problem is it works exactly as expected only on Windows, I've tested it on Windows 10 and 11, but it doesn't work on Linux, I've tested it on Ubuntu 22.04.
On Linux the ```App::update()``` function isn't called when there is no UI activity or window is minimized or in background.
I've replicated it in the [eframe template app](https://github.com/emilk/eframe_template) adding these three lines ad the end of the ```update()```:

```rust
println!("update() {}", count);
*count += 1;
ctx.request_repaint();
```
- On Windows 10 and 11 it works fine, you can see the uninterrupted stream of ```println()``` on the console even when the window is minimized or in background. 

- On Linux the printing freezes as soon there is no focus on the window or you put it in background.

Is there something I should do or know to achieve the expected behavior for a Linux target?

**Comment by wangxiaochuTHU** [comment]
You may want a secondary thread that runs *separately* from the UI thread. Here by *separately* I suggest that, in whatever case, the secondary thread should not be blocked by the UI thread, e.g, if the secondary thread wants to send something to the UI thread, nonblocking methods on nonblocking channels should be used.

**Comment by leklachu** [comment]
In case someone comes across this still (like I did just now! Thank you!), I'd like to mention putting `ctx.request_repaint();` inside the main `App::update()` function worked for me in Linux. The app kept running without mouse movement and also without focus.

**Reply by EdgeCaseBerg** [reply]
This is what I needed to do in order to get the window to repaint when I didn't have it in focus but was using a channel to send events to it in the background. I used `ctx.request_repaint_after(std::time::Duration::from_secs(1));` so that it wasn't constantly going off because a 1s refresh on the UI while it was in the background was good enough for my use case.
